src/components/victory-chart.jsx

```diff
@@ -881,157 +583,89 @@ export default class VictoryChart extends React.Component {
+      };
+    }
+    const data = !child.props.data && child.props.y ? this.generateData(child) : undefined;
+    return {
+      data,
+      animate,
+      domain: this.domain,
+      range: this.range,
+      scale: this.scale
     };
+  }
 
-    const dataByType = _.groupBy(this.datasets, (data) => {
-      return data.attrs.type;
-    });
-    // stackedBar and bar data must be treated as a group, and will
-    // be rendered in the order determined by the highest z-index
-    // in the group
-    const {stackedBar, bar, ...rest} = dataByType;
-    const stackedBarZ = stackedBar ? {
-      data: stackedBar,
-      z: _.max(_.map(stackedBar, "attrs.zIndex")) || this.style.bar.data.zIndex || 0,
-      type: "stackedBar"
-    } : [];
-    const barZ = bar ? {
-      data: bar,
-      z: _.max(_.map(bar, "attrs.zIndex")) || this.style.bar.data.zIndex || 0,
-      type: "bar"
-    } : [];
-
-    // the other data is dealt with independently
-    const otherData = _.flatten(_.values(rest));
-    const dataWithZ = _.isEmpty(otherData) ? [] :
-      _.map(otherData, (data) => {
-        const type = data.attrs.type;
-        const z = data.attrs.zIndex || this.style[type].data.zIndex || 0;
-        return {data, z, type};
-      });
-    // Put it all together, and sort the array of objects by z
-    const orderedDatasets = _.sortBy(dataWithZ.concat(stackedBarZ, barZ), "z");
-    return _.map(orderedDatasets, (dataset) => {
-      return functionMap[dataset.type].call(this, dataset.data);
+  // the old ones were bad
+  getNewChildren() {
+    return _.map(this.childComponents, (child, index) => {
+      const style = _.merge({}, {parent: this.props.style}, child.props.style);
+      const newProps = this.getNewProps(child);
+      return React.cloneElement(child, _.merge({}, newProps, {
+        ref: index,
+        key: index,
+        standalone: false,
+        style
+      }));
     });
   }
 
-  drawAxis(axis) {
-    const offsetY = axis === "y" ? undefined : this.axisOffset.y;
-    const offsetX = axis === "x" ? undefined : this.axisOffset.x;
-    const axisLabel = this.props.axisLabels && this.props.axisLabels[axis];
-    const animate = this.props.animate && (this.props.animate.axis || this.props.animate);
-    const style = _.merge({}, {parent: this.style.parent}, this.style.axis[axis]);
+  render() {
     return (
-      <VictoryAxis
-        {...this.props}
-        label={axisLabel}
-        animate={animate}
-        standalone={false}
-        offsetY={offsetY}
-        offsetX={offsetX}
-        crossAxis={true}
-        scale={this.scale[axis]}
-        domain={this.domain[axis]}
-        range={this.range[axis]}
-        orientation={this.props.axisOrientation[axis]}
-        tickCount={this.props.tickCount[axis]}
-        tickValues={this.tickValues[axis]}
-        tickFormat={this.tickFormat[axis]}
-        style={style}/>
+      <svg style={this.style.parent}>
+        {this.getNewChildren()}
+      </svg>
```

Comment:
Nice and small and clean now, but a feature was removed – seems like we might want to support rendering many charts into a single `<svg>`? But doesn't seem like a huge deal to me.